src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IepLevelService.kt

```diff
@@ -16,4 +17,56 @@ class IepLevelService(
       .toList()
       .sortedWith(compareBy(IepLevel::sequence))
   }
+
+  /**
+   * Because not all levels are enabled at all prisons, and we want to maintain a
+   * person’s level when they are transferred, we need to find the "nearest highest" level
+   * according to HMPPS policy. Falls back to prison's default level when there's nothing else to do.
+   */
+  suspend fun findNearestHighestLevel(prisonId: String, levelCode: String): String {
+    var maybeDefaultLevelCode: String? = null
+    val levelCodesAvailableInPrison = getIepLevelsForPrison(prisonId)
+      .filter { it.active }
+      .map {
+        if (it.default) {
+          maybeDefaultLevelCode = it.iepLevel
+        }
+        it.iepLevel
+      }
+    val defaultLevelCode = maybeDefaultLevelCode
+      // NOMIS traditionally takes the first if no default was found
+      ?: levelCodesAvailableInPrison.firstOrNull()
+      // there are no available incentive levels at all at this prison
+      ?: throw DataIntegrityException("$prisonId has no available incentive levels")
```

Comment:
serious data integrity problem: this prison has no incentive levels. you probably can't even assign a level to a person

Reply:
agreed